File: 696.count-binary-substrings/solution.go
func countBinarySubstrings(s string) int {
    p, q := 1, 0
    res := 0
    for i := 1; i < len(s); i++ {
        if s[i] == s[i - 1] {
            p++
        } else {
            p, q = 1, p
        }
        if q >= p {
            res++
        }
    }
    return res
}

File: 696.count-binary-substrings/solution.kt
class Solution {
    fun countBinarySubstrings(s: String): Int {
        var res = 0
        var p = 1
        var q = 0

        for (i in 1 until s.length) {
            if (s[i] == s[i - 1]) {
                p++
            } else {
                q = p
                p = 1
            }
            if (q >= p) {
                res++
            }
        }
        return res
    }
}